src/index.js

```diff
@@ -1098,42 +1098,36 @@ export function addAction (name, opts) {
     opts.response = opts.response || function (response) { return response }
     opts.responseError = opts.responseError || function (err) { return utils.reject(err) }
     mapper[name] = function (id, _opts) {
-      if (utils.isObject(id)) {
-        _opts = id
-      }
```

Comment:
Ah yeah, I was thinking it would just force the `getTeacherReportsAsync(null, { params: { foo: 'bar' } })` type of syntax - but that may be less appealing of an interface? Not sure on this one,  but if it's possible to avoid type-polymorphism in the arguments I think that would be good.